src/OrchardCore.Modules/OrchardCore.Search.Elasticsearch/Startup.cs

```diff
@@ -240,53 +197,88 @@ public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder ro
         {
-            var optionsAreValid = true;
-
             if (elasticConnectionOptions == null)
             {
                 logger.LogError("Elasticsearch is enabled but not active because the configuration is missing.");
                 return false;
             }
 
+            var optionsAreValid = true;
+
             if (string.IsNullOrWhiteSpace(elasticConnectionOptions.Url))
             {
                 logger.LogError("Elasticsearch is enabled but not active because the 'Url' is missing or empty in application configuration.");
                 optionsAreValid = false;
             }
 
-            if (elasticConnectionOptions.Ports.Length == 0)
+            if (elasticConnectionOptions.Ports?.Length == 0)
             {
                 logger.LogError("Elasticsearch is enabled but not active because a port is missing in application configuration.");
                 optionsAreValid = false;
             }
 
             return optionsAreValid;
         }
+
+        private static IConnectionPool GetConnectionPool(ElasticConnectionOptions elasticConfiguration)
+        {
+            var uris = elasticConfiguration.Ports.Select(port => new Uri($"{elasticConfiguration.Url}:{port}")).Distinct();
+            IConnectionPool pool = null;
+            switch (elasticConfiguration.ConnectionType)
+            {
+                case "SingleNodeConnectionPool":
+                    pool = new SingleNodeConnectionPool(uris.First());
+                    break;
+
+                case "CloudConnectionPool":
+                    if (!string.IsNullOrWhiteSpace(elasticConfiguration.Username) && !string.IsNullOrWhiteSpace(elasticConfiguration.Password) && !string.IsNullOrWhiteSpace(elasticConfiguration.CloudId))
+                    {
+                        var credentials = new BasicAuthenticationCredentials(elasticConfiguration.Username, elasticConfiguration.Password);
+                        pool = new CloudConnectionPool(elasticConfiguration.CloudId, credentials);
+                    }
+                    break;
+
+                case "StaticConnectionPool":
+                    pool = new StaticConnectionPool(uris);
+                    break;
+
+                case "SniffingConnectionPool":
+                    pool = new SniffingConnectionPool(uris);
+                    break;
+
+                case "StickyConnectionPool":
+                    pool = new StickyConnectionPool(uris);
+                    break;
+
+                default:
+                    pool = new SingleNodeConnectionPool(uris.First());
+                    break;
+            }
+
+            return pool;
+        }
     }
 
     [RequireFeatures("OrchardCore.Deployment")]
     public class DeploymentStartup : StartupBase
     {
         public override void ConfigureServices(IServiceCollection services)
         {
-            if (services.Any(d => d.GetImplementationType() == typeof(ElasticsearchService)))
```

Comment:
Ok that's the part that @sebastienros doesn't like. Then, I'm guessing we are removing this everywhere else too?